Source/cairo/CairoFontInfo.m

```diff
@@ -159,15 +202,38 @@ - (void) dealloc
 
 - (BOOL) glyphIsEncoded: (NSGlyph)glyph
 {
-  /* FIXME: There is no proper way to determine with the toy font API,
-     whether a glyph is supported or not. We will just ignore ligatures 
-     and report all other glyph as existing.
-  return !NSEqualSizes([self advancementForGlyph: glyph], NSZeroSize);
-  */
-  if ((glyph >= 0xFB00) && (glyph <= 0xFB05))
-    return NO;
-  else
-    return YES;
+  BOOL val = NO;
+
+  if (_scaled)
+    {
+      FT_Face face = cairo_ft_scaled_font_lock_face(_scaled);
+      if (FT_Get_Char_Index (face, glyph) != 0)
+	{
+	  val = YES;
+	}
+
+      cairo_ft_scaled_font_unlock_face(_scaled);
+    }
+  return val;
+}
+
+- (NSGlyph) glyphWithName: (NSString *)glyphName
+{
+  NSGlyph val = NSNullGlyph;
+  FT_String *name = (FT_String *)[glyphName UTF8String];
+
+  if (_scaled)
+    {
+      FT_Face face = cairo_ft_scaled_font_lock_face(_scaled);
+      if (FT_Has_PS_Glyph_Names(face))
+	{
+	  val = (NSGlyph) FT_Get_Name_Index(face, name);
+	}
+
```

Comment:
The superclass's implementation doesn't do anything useful, and the fallback is provided by knowing that FT_Get_Name_Index() returns zero when the named glyph is not present (or, really, for any error).

Reply:
OH, you meant for the first method, -glyphIsEncoded: -- The original method just returned YES for everything but five ligature glyphs, which I find extremely silly, while this version returns NO for everything that isn't a valid character in the font's encoding. I'm not sure what kind of fallback might be useful, since the font itself is the only thing that might know if it has a glyph at index N.